docs/_ext/utils.py

```diff
@@ -35,10 +35,41 @@ def readable_desc_rst(description):
     return ''.join(cleaned_lines)
 
 
+def readable_default(value, type_):
+    # default values of options are extracted from source file, but the
+    # representation in source code of a certain value does not necessarily
+    # identical to its representation in scylla.yaml, sometimes the default
+    # value is not even a literal. strictly speaking, it is an expression,
+    # so a function call is also valid. we have to translate them on a
+    # case-by-case basis
+    if type_ == 'tri_mode_restriction':
+        if value == 'strict_allow_filtering_default()':
+            # see db/config.cc
+            #   strict_allow_filtering_default()
+            return "warn"
```

Comment:
I would refrain from hardcoding values, as they might not be updated here if the code changes. Instead, we could consider leaving references to the functions (cleaning them if necessary to make sure they're readable). 

This approach might slightly impact user experience but ensures accurate, up-to-date documentation without the need to update the script constantly.

Additionally, we could explore ways to enhance its usefulness as part of a separate PR, such as linking to the code or better resolving the function.

Reply:
i understand the downside of this approach, but, IMHO, a regular user should not have the source code for using a software. it'd be difficult to address this without evaluating the function to get the value returned by it. anyway, i am extracting this commit into anther PR, and remove the part which is controversial.

Reply:
moved this change to #19566

Reply:
> i understand the downside of this approach, but, IMHO, a regular user should not have to read the source code for using a software. it'd be difficult to address this without evaluating the function to get the value returned by it. anyway, i am extracting this commit into anther PR, and remove the part which is controversial.

another way forwards is to actually compile and run this part of code into a tiny program which dumps the values of the option in a form which can be used directly in `scylla.yaml`, and parse the output of this program instead.